Jenkinsfile-dev-build

```diff
@@ -0,0 +1,258 @@
+                    resolver:
+                        repo: ecosys-dev-build-artifactory
+                        serverId: repo21
+                    deployer:
+                        repo: ecosys-dev-build-artifactory
+                        serverId: repo21
+                """
+                sh 'mkdir -p .jfrog/projects'
+                writeFile(file: '.jfrog/projects/go.yaml', text: yamlContent)
+                echo "Successfully created .jfrog/projects/go.yaml"
+
+                stage('Last Release Build') {
+                    downloadJfrogCli()
+                    sh "${builderPath} --version"
+                }
+
+                stage('Clone Source Repo') {
+                    dir("${cliWorkspace}") {
+                        sh "git clone ${featureRepo}"
+                        dir("${repo}"){
+                            sh "git checkout ${featureBranch}"
+                        }
+                    }
+                }
+
+                stage('jf build and upload binaries phase') {
+                    buildAndUploadBinaryAndBuildInfo(architectures)
+                }
+            }
+        }
+    }
+}
+
+def downloadJfrogCli() {
+    script {
+        def cliVersion = "[RELEASE]"
+        def cliOs = ""
+        def cliArch = ""
+        def fileName = "jf"
+
+        def osName = System.getProperty("os.name").toLowerCase()
+        def osArch = System.getProperty("os.arch").toLowerCase()
+
+        echo "Detected OS Name: '${osName}', Arch: '${osArch}'"
+
+        if (osName.contains("linux")) {
+            cliOs = "linux"
+        } else if (osName.contains("mac")) {
+            cliOs = "mac"
+        } else if (osName.contains("windows")) {
+            cliOs = "windows"
+            fileName = "jf.exe"
+        } else {
+            error("Unsupported Operating System: ${osName}")
+        }
+
+        switch(osArch) {
+            case "amd64":
+            case "x86_64":
+                cliArch = "amd64"
+                break
+            case "aarch64":
+                cliArch = "arm64"
+                break
+            case "i386":
+            case "x86":
+                cliArch = "386"
+                break
+            default:
+                error("Unsupported Architecture: ${osArch}")
+                break
+        }
+
+        def downloadUrl = "https://releases.jfrog.io/artifactory/jfrog-cli/v2-jf/${cliVersion}/jfrog-cli-${cliOs}-${cliArch}/${fileName}"
+
+        sh """
+            #!/bin/bash
+            set -ex
+            curl -fL -o '${builderPath}' '${downloadUrl}'
+            chmod +x '${builderPath}'
+        """
+        echo "JFrog CLI downloaded successfully as '${fileName}' in the current workspace."
+    }
+}
+
+def getCliVersion(exePath) {
+    version = sh(script: "$exePath -v | tr -d 'jfrog version' | tr -d '\n'", returnStdout: true) + "-beta"
+    return version
+}
+
+def buildAndUploadBinaryAndBuildInfo(architectures) {
+    version = getCliVersion(builderPath)
```

Comment:
doesn't this give previous release version with -beta at end? we might need to have a beta with next possible version? WDYT?